File: examples/datakeep_src_data/output.cs
using System;
namespace Output 
{
struct PStruct
{
public string name;
public PTag[] tags;
public PField[] fields;
public string inheritance;
public static string ToString(PStruct struct_)
{
string result = "";
result += struct_.name;
foreach( PTag e in struct_.tags) {
result +=  PTag.ToString(e);
}
foreach( PField e in struct_.fields) {
result +=  PField.ToString(e);
}
result +=  string.ToString(e);
}
}
struct PField
{
public string name;
public PTag[] tags;
public string type;
public static string ToString(PField struct_)
{
string result = "";
result += struct_.name;
foreach( PTag e in struct_.tags) {
result +=  PTag.ToString(e);
}
result += struct_.type;
}
}
struct PTag
{
public string name;
public string[] arguments;
public static string ToString(PTag struct_)
{
string result = "";
result += struct_.name;
foreach( string e in struct_.arguments) {
result +=  string.ToString(e);
}
}
}
}

File: examples/datakeep_src_data/output.py
from datakeep import DataKeepStruct, DataKeepField, DataKeepTag
DEFS = {
"s" : "string",
"s[]" : "string[]"
}

FLAGS = {
}

ALL_STRUCTS = []
ALL_STRUCTS.append(DataKeepStruct("Base", [DataKeepTag("NoBody", [])], [DataKeepField("name", "string", []),DataKeepField("tags", "PTag[]", [DataKeepTag("Convert", ["arr", " PTag"])])]))
ALL_STRUCTS.append(DataKeepStruct("PStruct", [DataKeepTag("ToString", []),DataKeepTag("Print", [])], [DataKeepField("name", "string", []),DataKeepField("tags", "PTag[]", [DataKeepTag("Convert", ["arr", " PTag"])]),DataKeepField("fields", "PField[]", [DataKeepTag("Convert", ["arr", " PField"])]),DataKeepField("inheritance", "string", [])]))
ALL_STRUCTS.append(DataKeepStruct("PField", [DataKeepTag("ToString", []),DataKeepTag("Print", [])], [DataKeepField("name", "string", []),DataKeepField("tags", "PTag[]", [DataKeepTag("Convert", ["arr", " PTag"])]),DataKeepField("type", "string", [])]))
ALL_STRUCTS.append(DataKeepStruct("PTag", [DataKeepTag("ToString", []),DataKeepTag("Print", [])], [DataKeepField("name", "string", []),DataKeepField("arguments", "string[]", [DataKeepTag("Convert", ["arr", " string"])])]))
